### PR TITLE
Add CISPO and SAPO loss type support for Triton GRPO loss kernel

### PR DESCRIPTION
**Pull request by yukiu00**
## Summary

Add **CISPO** and **SAPO** loss type support to the Triton `ops/grpo_loss.py` kernel.

This is a follow-up to:
- #1054 - Add CISPO loss type support for LigerFusedLinearGRPOLoss (chunked loss path)
- #1073 - Add SAPO loss type support for LigerFusedLinearGRPOLoss (chunked loss path)

> **Note**: This PR depends on #1073 (SAPO PR) being merged first, as it builds on top of that branch.

### Background

PR #1054 and #1073 added CISPO and SAPO support to the `chunked_loss` path, but the `ops` (Triton kernel) path was marked as a follow-up. This PR implements that follow-up.

### Changes

**`src/liger_kernel/ops/grpo_loss.py`**
- Add loss type constants (`_LOSS_TYPE_GRPO`, `_LOSS_TYPE_CISPO`, `_LOSS_TYPE_SAPO`) with `tl.constexpr` for compile-time branching
- Implement CISPO in forward/backward kernels:
  - Upper-bound only clipping (no lower bound)
  - Detached coefficient (gradient only flows through logp)
  - Loss formula: `-coef_2 * advantage * logp`
- Implement SAPO in forward/backward kernels:
  - Sigmoid-based soft gating instead of hard clipping
  - Different temperatures for positive/negative advantages
  - Loss formula: `-sigmoid(τ*(ρ-1)) * 4/τ * advantage`
- Update `GrpoLossFunction` with `loss_type`, `sapo_temperature_pos`, `sapo_temperature_neg` parameters

**`src/liger_kernel/transformers/grpo_loss.py`**
- Remove error blocking for CISPO and SAPO loss types
- Add `sapo_temperature_pos` and `sapo_temperature_neg` parameters
- Update `_reduce_grpo_loss` to handle CISPO (DAPO normalization) and SAPO (GRPO normalization)

**`test/transformers/test_grpo_loss.py`**
- Add reference PyTorch implementations (`torch_cispo_loss`, `torch_sapo_loss`)
- Add `test_cispo_loss` and `test_sapo_loss` test functions

## Testing Done

- Hardware Type: NVIDIA GPU
- [x] run `make test` to ensure correctness
- [x] run `make checkstyle` to ensure code style
- [ ] run `make test-convergence` to ensure convergence

### CODE REVIEW

**Comment by Tcc0403** [comment]
Please resolve conflict, thanks.

**Comment by yukiu00** [comment]
@Tcc0403 

I've resolved the conflicts! Could you review this PR?

File: test/transformers/test_grpo_loss.py [hunk 1]
```diff
@@ -188,3 +279,151 @@ def test_grpo_loss(B, T, V, temperature, num_iteration, beta, eps_low, eps_high,
     compare(loss2, loss3, "per_token_loss: triton-bf16 vs torch-fp32, ")
     compare(kl2, kl3, "per_token_kl: triton-bf16 vs torch-fp32, ")
     compare(logits2.grad, logits3.grad, "logits.grad: triton-bf16 vs torch-fp32, ")
+
```
Comment (Tcc0403) [inline test/transformers/test_grpo_loss.py]: I just realized we never assert in grpo test. Use either `[torch.testing.assert_close()](https://docs.pytorch.org/docs/stable/testing.html)` or `utils.assert_verbose_allclose()`
https://github.com/linkedin/Liger-Kernel/blob/ad6f0a74922e4a3130a9926f82b2c18159a29911/test/utils.py#L90

Comment (Tcc0403) [reply test/transformers/test_grpo_loss.py]: I did check the stdout of this test and results are all close, so the kernel implementations should be correct.

File: test/transformers/test_grpo_loss.py
Comment (Tcc0403) [inline test/transformers/test_grpo_loss.py]: We can remove it then

Comment (Tcc0403) [inline test/transformers/test_grpo_loss.py]: Can you fix this part as well?

**Comment by yukiu00** [comment]
@Tcc0403 Thanks for the review! I've addressed the feedback:

- Removed the `compare()` function that only printed differences without asserting
- Replaced all `compare()` calls with `assert_verbose_allclose()` across all 4 test functions (`test_selective_log_softmax`, `test_grpo_loss`, `test_cispo_loss`, `test_sapo_loss`)
- Set tolerances to `atol=5e-2, rtol=5e-1` to match `test/chunked_loss/test_grpo_loss.py`
- Added null guards for KL comparisons (`beta == 0.0` case)

All 14 tests pass. Please take another look!
